src/nvim/api/ui.c

```diff
@@ -512,12 +512,13 @@ void nvim_ui_pum_set_bounds(uint64_t channel_id, Float width, Float height, Floa
   ui->pum_pos = true;
 }
 
-/// Tells Nvim when a terminal event has occurred: sets |v:termresponse| and fires |TermResponse|.
+/// Tells Nvim when a terminal event has occurred
 ///
 /// The following terminal events are supported:
 ///
-///   - "osc_response": The terminal sent a OSC response sequence to Nvim. The
-///                     payload is the received OSC sequence.
+///   - "termresponse": The terminal sent an OSC or DCS response sequence to
```

Comment:
first parameter could be "type", then we can pass that as the event pattern

Reply:
I'm not convinced we need this. The "type" of the sequence can be discerned from the sequence itself (e.g. a DCS always begins with `ESC P`). We will always include the full sequence (including the OSC/DCS prefix) so that autocommands have the ability to handle both OSCs and DCS sequences and distinguish between them. So including the "type" in the pattern doesn't provide any more information.

The TermResponse autocommand is rare and low frequency, so being able to filter out unwanted events is not that much of a benefit either.

Reply:
Is it a one-way door? I.e. will it be hard to change this event signature later?

Reply:
Probably not. If you feel strongly about it, we can do it. But I really don't see the benefit -- it's additional complexity (in both code and documentation) that doesn't pay for itself imo.

Reply:
I don't feel strongly about it, I thought it was something we already had and is easy to pass along.

Does this mean it's also not worth doing for TermRequest?

Reply:
>Does this mean it's also not worth doing for TermRequest?

I think so, for the same reasons I mentioned here. Anyone that wants to handle a TermRequest autocommand already needs to know how to parse the escape sequence, and in doing so it is trivially easy to determine if it's an OSC or DCS from the sequence itself.

>OTOH TermRequest could be very frequent, right? E.g. for decorating URLs.

Maybe, I'm not as familiar with the use case for TermRequest.